samples/subsys/usb/hid/src/main.c

```diff
@@ -61,60 +70,75 @@ static const uint8_t hid_report_desc[] = {
 
 static void send_report(struct k_work *work)
 {
-	static uint8_t report_1[2] = { REPORT_ID_1, 0x00 };
 	int ret, wrote;
 
-	ret = hid_int_ep_write(hdev, report_1, sizeof(report_1), &wrote);
-
-	LOG_DBG("Wrote %d bytes with ret %d", wrote, ret);
-
-	/* Increment reported data */
-	report_1[1]++;
+	if (!atomic_test_and_set_bit(hid_ep_in_busy, HID_EP_BUSY_FLAG)) {
+		ret = hid_int_ep_write(hdev, (uint8_t *)&report_1,
+				       sizeof(report_1), &wrote);
+		if (ret != 0) {
+			LOG_ERR("Failed to submit report");
+		} else {
```

Comment:
Shouldn't `hid_ep_in_busy` be cleared when the write failed? `int_in_ready_cb` will not be called in such case, right? Or is it expected that no more reports will be sent until the device is configured again (then a comment could help)?

Reply:
> Shouldn't `hid_ep_in_busy` be cleared when the write failed? `int_in_ready_cb` will not be called in such case, right?

Actually there is no common/well-tested way to recover from that. Bad excuse: very unlikely that error will happen.

Reply:
comment added